src/features/user/ManageProjects/ProjectsContainer.tsx

```diff
@@ -127,6 +128,7 @@ export default function ProjectsContainer() {
   // This effect is used to get and update UserInfo if the isAuthenticated changes
   React.useEffect(() => {
     if (contextLoaded && token) {
+      localStorage.setItem('redirectLink', router.asPath);
       loadProjects();
```

Comment:
I don't particularly like that we are setting a local storage value to detect where the user came to the project details page from.

There could be issues with this approach - perhaps the local storage value gets set when this page is initially visited, but through internal navigation in the app the user accesses another page which links to a project. We'll need to ensure the value is up to date whenever such a page is visited, otherwise the back navigation may take the user to an unexpected place.

Would it be more reliable to pass the `backNavigationUrl` as a query param to the project details page? That could simplify the logic, we'd only need to take the user to the `backNavigationUrl` if available, or to the the project list page (default).

Reply:
The query param logic is the good idea

Reply:
Sounds good, keep this query param specific to the project details page only.